app/src/main/java/org/simple/clinic/editpatient/EditPatientEffectHandler.kt

```diff
@@ -54,6 +59,19 @@ class EditPatientEffectHandler @AssistedInject constructor(
         .build()
   }
 
+  private fun fetchAllPrefillForm(scheduler: Scheduler) =
+      ObservableTransformer<FetchBangladeshNationalIdEffect, EditPatientEvent> { effectStream ->
+        effectStream
+            .flatMap { patientRepository.bangladeshNationalIdForPatient(it.patient.uuid).subscribeOn(scheduler) }
+            .map {
+              when (it) {
+                is Just -> it.value.identifier.value
+                None -> ""
+              }
+            }
```

Comment:
There is already a utility function called `unwrapJust` which just gives you the `Just` from an `Optional`.
You won't need this `when` block.

Reply:
This change has been done in the following commits.